tests/frame/getitem_test.py

```diff
@@ -193,3 +195,58 @@ def test_slice_edge_cases(constructor_eager: Any) -> None:
     assert df[[], ["a"]].shape == (0, 1)
     assert df[:, :].shape == (3, 4)
     assert df[[], []].shape == (0, 0)
+
+
+@pytest.mark.parametrize(
+    ("row_idx", "col_idx"),
+    [
+        ([0, 2], [0]),
+        ((0, 2), [0]),
+        ([0, 2], (0,)),
+        ((0, 2), (0,)),
+        ([0, 2], range(1)),
+        (range(2), [0]),
+        (range(2), range(1)),
+    ],
+)
+def test_get_item_works_with_tuple_and_list_indexing(
+    constructor_eager: Any,
+    row_idx: list[int] | tuple[int] | range,
```

Comment:
@Riik @windiana42 We can also type hint this as a `Sequence[int]`, correct?

Reply:
Yeah I think that is cleaner! Since 'Sequence' is what we were trying to support

Reply:
On second thought, I am not sure we should because `Sequence` includes strings. We would want a `SequenceNotStr` type hint, which I don't think we want to introduce now.